### PR TITLE
Updates per PRs 6044, 6057, 6068 

### PR DESCRIPTION
**Pull request by peiying2**

### CODE REVIEW

**Comment by flodolo** [comment]
I won't go through the files until Monday, but there are 2 glaring errors:
* download_buttons.lang is a shared file, like main.lang. If you add a string there, it's already available to other templates, and no point in adding it a second time.
* have you tried looking on Transvision for the strings in macros.html? Clearly we don't need a new file, when the PR it's only adding 2 strings.

File: en-US/firefox/all.lang [hunk 1]
```diff
@@ -74,6 +74,10 @@ Thanks to contributions of Mozilla community members around the world, Firefox i
 <strong>Warning</strong>: Nightly is updated daily. It is most appropriate for core Mozilla contributors and early adopters, not regular users.
 
 
+;<a href="%(url)s">Mozilla no longer provides security updates for Firefox on Windows XP or Vista</a>, but you can still download the final Windows 32-bit version below.
```
Comment (flodolo) [inline en-US/firefox/all.lang]: This string is already available in a shared file, you don't need it duplicated here.

File: en-US/macros.lang [hunk 2]
```diff
@@ -0,0 +1,146 @@
+## NOTE: https://www-demo5.allizom.org/
```
Comment (flodolo) [inline en-US/macros.lang]: Drop this file, and add to main.lang strings that are not available from the [new piece of code](https://github.com/jpetto/bedrock/blob/4a4da9d750695cc0c592690773d9874aa43438c5/bedrock/base/templates/macros.html#L410-L441).

File: en-US/mozorg/newsletters.lang [hunk 3]
```diff
@@ -65,6 +66,11 @@ Open your inbox (and your heart) even more — take a look at other topics we co
 Love the web? So do we!
 
 
+## TAG: developer_page_desc_082018
+;Unlock the world of web development with our weekly Mozilla Developer Newsletter. Each edition brings you coding techniques and best practices, MDN updates, info about emerging technologies, developer tools tips, and more.
+Unlock the world of web development with our weekly Mozilla Developer Newsletter. Each edition brings you coding techniques and best practices, MDN updates, info about emerging technologies, developer tools tips, and more.
+
+
 ;Join thousands of developers like you who are learning the best of web development.
```
Comment (flodolo) [inline en-US/mozorg/newsletters.lang]: Add obsolete comment

File: en-US/macros.lang (hunk 2 above)
Comment (peiying2) [reply en-US/macros.lang]: I will add the strings to main.lang. I did see strings available in many files, and was thinking about using your script to leverage them. Ok. I will move new strings there.

File: en-US/firefox/all.lang (hunk 1 above)
Comment (peiying2) [reply en-US/firefox/all.lang]: Weir, I can't find it in transvision but search in VM pointed to the right place.

**Comment by flodolo** [comment]
> My comment was pointing to a specific set of lines (strings) added in the PR (for macros), not all strings in the file.

To clarify this point even further: macros.html contains a bunch of strings. Some of them are used in sendto.lang (which is not shared), others we don't care about, the new ones needs to go into a shared file (main.lang).

File: en-US/firefox/all.lang (hunk 1 above)
Comment (flodolo) [reply en-US/firefox/all.lang]: How can Transvision find a string that is not available yet in the repository?

File: en-US/main.lang [hunk 4]
```diff
@@ -817,3 +817,39 @@ Create account
...
+
+;Retweet
+Retweet
+
+
+;Favorite
+Favorite
+
+
+;<strong>Enter your email</strong> to access Firefox Accounts.
```
Comment (flodolo) [inline en-US/main.lang]: Remove all new strings above this, they're not part of [this range](https://github.com/jpetto/bedrock/blob/4a4da9d750695cc0c592690773d9874aa43438c5/bedrock/base/templates/macros.html#L410-L441).

**Comment by flodolo** [comment]
This looks good. Given this PR is already several days old, I'd suggest to:

a) Run gitup, make sure that trunk and prod are in sync

Only at that point, switch back to your branch and:
a) rebase against master
b) add all other languages
c) merge and immedialy cherry pick to production

Let me know if you need help with this.

**Pull request merged**